oap-server/ai-pipeline/src/main/java/org/apache/skywalking/oap/server/ai/pipeline/services/HttpUriRecognitionService.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.ai.pipeline.services;
+
+import java.util.List;
+import org.apache.skywalking.oap.server.ai.pipeline.services.api.HttpUriRecognition;
+
+public class HttpUriRecognitionService implements HttpUriRecognition {
+    @Override
+    public void recognize(final String serviceId, final List<HTTPUri> unrecognizedURIs, final Callback callback) {
+
```

Comment:
This is TODO, to build the connection between OAP and the remote server.

Reply:
@Superskyyy This is for the whole process. `serviceId` and `unrecognizedURIs` are the raw data for you. `callback` will return the patterns to the kernel. Nothing more.

I feel you are being held by your old design. I am repeating myself again, runtime matching(including further metrics baseline calculation for alerting) is not an AI part responsibility.